src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java

```diff
@@ -86,6 +90,31 @@ public Agency getAgency(final String agencyId, final StatusFilter filter, final
                 .map(AgencyTransformer::transform).orElseThrow(EntityNotFoundException.withId(agencyId));
     }
 
+
+    @Transactional
+    public Agency updateAgency(final String agencyId, final Agency agencyToUpdate) {
+        final var agency = agencyLocationRepository.findById(agencyId).orElseThrow(EntityNotFoundException.withId(agencyId));
+        if (!agency.getId().equalsIgnoreCase(agencyToUpdate.getAgencyId())) {
```

Comment:
Should we have a different type for updates, feels a bit weird to specify the id in both the path and payload? Or is this a pattern we use else where?

Reply:
Interestingly there might even have been an argument for a Prison, Court, ProbationOffice sub types since I suspect this is how we might represent different types on locations in some independent RefData, I bet a Court will end up with different attributes to a Prison etc ..

Reply:
True, might put this as next PR